### PR TITLE
docs: update release process for one-off packages

### LINKED ISSUE #3544: Update docs on releasing individual packages
I accidentally released `modal` without `shared` since we document to use `yarn version` for individual packages. We need to revisit this and how to properly do this with checking dependency packages.

### PR DESCRIPTION
**Pull request by bryceosterhaus**
fixes #3544

This is just a general idea for now. I think ideally we could find a more automated process for releasing one-off packages.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -229,10 +229,15 @@ In a rare case you may want to release only one specific package. To do so, foll
 
 1. `git checkout master`
 2. Navigate to specific package (`cd ./packages/$COMPONENT_DIR`)
-3. Run `yarn version` and choose what the new version will be.
-4. Create a draft PR (not intended to be merged) to the clay repo; the sole purpose of this is to see CI green one last time before pushing the tag.
-5. Once CI is green, run `git push $REMOTE master --follow-tags`
-6. Run `yarn publish` to publish module to the npm registry.
+3. Look at the git history of the package and determine if it relies on any non-released code from other packages.
+4. If it relies on other packages, **you must publish those dependencies first** following these same steps(start at step 1 with the dependent package).
```
Comment (wincent) [inline CONTRIBUTING.md:233]: Missing space before "(".

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -229,10 +229,15 @@ In a rare case you may want to release only one specific package. To do so, foll
...
+
+_After you have published any dependencies needed_
+
+5. Run `yarn version` and choose what the new version will be. Any _feat_ commits since last release require a minor version. If only _chore_ or _fix_ commits, a patch is required.
+6. Create a draft PR (not intended to be merged) to the clay repo; the sole purpose of this is to see CI green one last time before pushing the tag.
+7. Once CI is green, run `git push $REMOTE master --follow-tags`
```
Comment (wincent) [inline CONTRIBUTING.md:239]: For the paranoid, could mention `git push $REMOTE master --follow-tags --dry-run` to preview what will be pushed. That has saved me from pushing junk tags on occasion.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -229,10 +229,15 @@ In a rare case you may want to release only one specific package. To do so, foll
 
 1. `git checkout master`
 2. Navigate to specific package (`cd ./packages/$COMPONENT_DIR`)
-3. Run `yarn version` and choose what the new version will be.
-4. Create a draft PR (not intended to be merged) to the clay repo; the sole purpose of this is to see CI green one last time before pushing the tag.
-5. Once CI is green, run `git push $REMOTE master --follow-tags`
-6. Run `yarn publish` to publish module to the npm registry.
+3. Look at the git history of the package and determine if it relies on any non-released code from other packages.
```
Comment (wincent) [inline CONTRIBUTING.md:232]: simple

File: CONTRIBUTING.md (hunk 2 above)
Comment (bryceosterhaus) [reply CONTRIBUTING.md:239]: I didn't even know that flag existed, right on!

**Pull request merged**